### ISSUE TITLE
`rustc` warns when finding an unknown jobserver style

### ISSUE THREAD

**Issue by ojeda** [rust-lang/rust#120532]
`rustc` warns when finding an unknown jobserver style, but the Make manual seems to suggest to ignore it (since new kinds may be introduced in the future), or at least that is one possible interpretation:

> If your tool does not recognize the format of the `--jobserver-auth` string, it should assume the jobserver is using a different style and it cannot connect.

One could also read that as "... and then warn about it", like it asks to do for the simple pipe case. But if that is the case, I am not sure why it would say "it cannot connect" since obviously one would not be able to do so, i.e. I take it to mean "ignore it".

Thus, should `CannotParse` from `jobserver` be taken as an OK case, or should we keep the current behavior?

```console
$ echo 'fn main() {}' | MAKEFLAGS=--jobserver-auth=new rustc -
warning: failed to connect to jobserver from environment variable `MAKEFLAGS="--jobserver-auth=new"`: cannot parse jobserver environment variable value: expected `fifo:PATH` or `R,W`, found `new`
  |
  = note: the build environment is likely misconfigured
```

From https://github.com/rust-lang/rust/issues/120515.

**Comment by petrochenkov** [comment]
I'd prefer to keep the current behavior, if some specific `--jobserver-auth` format actually exists and is used, then jobserver-rs should support it, and the we won't report a warning in that specific case.

**Comment by ojeda** [comment]
Yeah, though users may have updated GNU Make and not `rustc`. Now, that may be unlikely, and it requires a new jobserver style to being with, but it could happen. But, in that case, I would expect users to force their GNU Make to use an older jobserver style if they are really the kind of project that cannot upgrade `rustc`.
